File: examples/managed-nodegroups-cs/Iam.cs
using Aws = Pulumi.Aws;

static class Iam
{
    private static readonly string[] s_managedPolicyArns =
    {
        "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
        "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
        "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
    };

    /// <summary>
    /// Creates a role and attaches the EKS worker node IAM managed policies.
    /// </summary>
    public static Aws.Iam.Role CreateRole(string name)
    {
        var role = new Aws.Iam.Role(name, new Aws.Iam.RoleArgs
        {
            AssumeRolePolicy = @"{
""Version"": ""2008-10-17"",
""Statement"": [{
    ""Sid"": """",
    ""Effect"": ""Allow"",
    ""Principal"": {
        ""Service"": ""eks.amazonaws.com""
    },
    ""Action"": ""sts:AssumeRole""
}]
}",
        });

        for (int i = 0; i < s_managedPolicyArns.Length; i++)
        {
            // Create RolePolicyAttachment without returning it.
            new Aws.Iam.RolePolicyAttachment($"{name}-policy-{i}", new Aws.Iam.RolePolicyAttachmentArgs
            {
                PolicyArn = s_managedPolicyArns[i],
                Role = role.Id,
            });
        }

        return role;
    }
}

File: examples/nodegroup-py/iam.py
import json
import pulumi_aws as aws

managed_policy_arns = [
    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
]

# Creates a role and attaches the EKS worker node IAM managed policies
def create_role(name: str) -> aws.iam.Role:
    role = aws.iam.Role(name, assume_role_policy=json.dumps({
        "Version": "2012-10-17",
        "Statement": [
            {
                "Sid": "AllowAssumeRole",
                "Effect": "Allow",
                "Principal": {
                    "Service": "ec2.amazonaws.com",
                },
                "Action": "sts:AssumeRole"
            }
        ]
    }))

    for i, policy in enumerate(managed_policy_arns):
        # Create RolePolicyAttachment without returning it.
        rpa = aws.iam.RolePolicyAttachment(f"{name}-policy-{i}",
            policy_arn=policy,
            role=role.id)

    return role
